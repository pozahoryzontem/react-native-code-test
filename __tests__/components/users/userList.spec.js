import React from 'react';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { FlatList } from 'react-native';
import thunk from 'redux-thunk';
import UserList from '../../../src/components/users/userList';


describe('UserList', () => {
  it('should show users', () => {
    const mockStore = configureMockStore();
    const initialState = {
      users: [
        {
          id: 0,
          firstName: 'Michael',
          lastName: 'Lawson',
          email: 'michael.lawson@reqres.in',
          avatar: 'https://s3.amazonaws.com/uifaces/faces/tw4ditter/follettkyle/128.jpg',
        },
      ],
      alert: {},
      pageNumber: 1,
      canLoadUsers: true,
    };

    const store = mockStore(initialState);
    const component = create(
      <UserList store={store} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  // I got stuck here
  it.skip('should call getUsersConnect when onEndReached', () => {
    const initialState = {
      users: [
        {
          id: 0,
          firstName: 'Michael',
          lastName: 'Lawson',
          email: 'michael.lawson@reqres.in',
          avatar: 'https://s3.amazonaws.com/uifaces/faces/tw4ditter/follettkyle/128.jpg',
        },
      ],
      alert: {},
      pageNumber: 1,
      canLoadUsers: true,
    };
    const mockStore = configureMockStore([thunk]);
    const store = mockStore(initialState);

    const mockGetUsersConnect = jest.fn();
    const userListRender = create(<UserList store={store} getUsersConnect={mockGetUsersConnect} />);

    const instance = userListRender.root;
    const flatList = instance.findByType(FlatList);
    flatList.props.onEndReached();
    expect().toBeCalled();
  });
});
