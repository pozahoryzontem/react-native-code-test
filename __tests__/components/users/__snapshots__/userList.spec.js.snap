// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList should show users 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 10,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      Array [
        Object {
          "avatar": "https://s3.amazonaws.com/uifaces/faces/tw4ditter/follettkyle/128.jpg",
          "email": "michael.lawson@reqres.in",
          "firstName": "Michael",
          "id": 0,
          "lastName": "Lawson",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.0001}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "marginBottom": 10,
              "marginTop": 10,
            }
          }
        >
          <Image
            onError={[Function]}
            source={
              Object {
                "uri": "https://s3.amazonaws.com/uifaces/faces/tw4ditter/follettkyle/128.jpg",
              }
            }
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": 40,
                "height": 80,
                "marginRight": 20,
                "width": 80,
              }
            }
          />
          <Text
            style={
              Object {
                "fontFamily": "Cochin",
                "fontSize": 20,
              }
            }
          >
            Michael Lawson
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
------------HTML PREVIEW---------------
<View style="flex:1;padding-top:10px;">
  <RCTScrollView>
    <View>
      <View>
        <View style="align-items:center;flex:1;flex-direction:row;margin-bottom:10px;margin-top:10px;">
          <Image style="background-color:white;border-radius:40px;height:80px;margin-right:20px;width:80px;"></Image>
          <Text style="font-family:Cochin;font-size:20px;">
            Michael Lawson
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
